com.unity.render-pipelines.core/Editor/RenderPipelineConverter.cs

```diff
@@ -0,0 +1,196 @@
+    /// WarningMessage = If there are some issues with the converter that we already know about.
+    ///     Example: If we know it is a custom shader, we can not convert it so we add the information here.
+    /// HelpLink = Link to the documentation of how to convert this asset. Useful if the conversion failed or if we know we can not convert this asset automatically.
+    /// </summary>
+    public struct ConverterItemDescriptor
+    {
+        public string name;
+        public string info;
+        public string warningMessage;
+        public string helpLink;
+    }
+
+    /// <summary>
+    /// A structure holding the information for each Item that needs to be Converted.
+    /// Descriptor = The ConverterItemDescriptor this item contain.
+    /// Index = The index for this item in the list of converter items.
+    /// </summary>
+    public struct ConverterItemInfo
+    {
+        public ConverterItemDescriptor descriptor { get; internal set; }
+        public int index { get; internal set; }
+    }
+
+    // Storing the index and message of the failed item so that we can show that in the UI.
+    internal struct FailedItem
+    {
+        public int index;
+        public string message;
+    }
+
+    internal struct SuccessfulItem
+    {
+        public int index;
+    }
+
+    /// <summary>
+    /// A structure needed for the initialization step of the converter.
+    /// Stores data to be visible in the UI.
+    /// </summary>
+    public struct InitializeConverterContext
+    {
+        /// <summary>
+        /// Stores the list of ConverterItemDescriptor that will be filled in during the initialization step.
+        /// </summary>
+        internal List<ConverterItemDescriptor> items;
+
+        /// <summary>
+        /// Add to the list of assets to be converted.
+        /// This will be used to display information to the user in the UI.
+        /// </summary>
+        /// <param name="item">The item to add to the list items to convert</param>
+        public void AddAssetToConvert(ConverterItemDescriptor item)
+        {
+            items.Add(item);
+        }
+    }
+
+    /// <summary>
+    /// A structure needed for the conversion part of the converter.
+    /// This holds the index for failed and successful items converted.
+    /// </summary>
+    public struct RunConverterContext
+    {
+        List<ConverterItemInfo> m_Items;
+
+        internal List<FailedItem> m_FailedItems;
+        internal List<SuccessfulItem> m_SuccessfulItems;
+
+        /// <summary>
+        /// These are the items that the converter needs to iterate over.
+        /// The UI will make this list of items only the ones that are ticked and should be converted.
+        /// </summary>
+        public IEnumerable<ConverterItemInfo> items => m_Items;
+
+        internal int failedCount => m_FailedItems.Count;
+        internal int successfulCount => m_SuccessfulItems.Count;
+
+        internal FailedItem GetFailedItemAtIndex(int index)
```

Comment:
What are these needed for if `m_FailedItems` and `m_SuccessfulItems` are `internal` too?

Reply:
made them private

Reply:
done